sdk/rpc/server/server.go

```diff
@@ -145,6 +146,11 @@ func (t *RPCServer) CredentialProvisionerDescription(req proto.ProvisionerID, re
 // interface. The call is forwarded to the Provision() function of the Provisioner of the credential identified by
 // req.CredentialID.
 func (t *RPCServer) CredentialProvisionerProvision(req proto.ProvisionCredentialRequest, resp *sdk.ProvisionOutput) error {
+	defer func() {
+		if err := recover(); err != nil {
+			resp.AddError(fmt.Errorf("your locally built plugin failed with the following panic: %s; and with stack trace: %s", err, string(debug.Stack())))
+		}
+	}()
```

Comment:
before attempting to put "panic catchers" in the other functions as well, I figured a discussion about which technique we'd like to use here should be in place @jpcoenen

Reply:
I think the panic is part of the RPC layer and should therefore not "leak" into the rest of the plugin system. Also, an alternative method could provide some more flexibility for the CLI to deal with these panics.

What if instead of adding it to the `sdk.ProvisionOutput` type, we instead wrap the `sdk.ProvisionOutput` type in `proto.ProvisionCredentialResponse` type. For example:

```
type Panic struct {
       Error string
       Stack []byte 
}

type ProvisionCredentialResponse struct {
     Success sdk.ProvisionOutput
     Panic *Panic
}
```

Small downside of this is that we explicitly have to make sure to check if `Panic == nil` in the CLI, but I think that outweighs the fact that we can better tell if the plugin is panicking.

What do you think?

Reply:
Indeed the panic we are trying to catch does not fall into the same category as a plugin error that we can handle in the CLI but rather a RPC specific error that we need to handle at the RPC layer.

Would the proposed solution then entail changing the RPC functions' signature to 
`CredentialProvisionerProvision(req proto.ProvisionCredentialRequest, resp proto.ProvisionCredentialResponse)`, and therefore would require us to bump up the protocol version? @jpcoenen

Reply:
Yes, good call-out, you're right. I am not sure if it this is by itself is worth bumping the protocol version. 

We could also go for a dual approach: apply your suggestion to `main` now and also merge an improved method to the `protocol-v4` branch and leave it there until we have another change that also needs a protocol version bump. Then we immediately improve the status quo without breaking anything, but also have a nicer solution long-term. WDYT?

Reply:
The dual approach sounds good! I have reflected this plan in our internal issue: https://gitlab.1password.io/dev/b5/op/-/issues/3321. 

I will update all RPC functions with the simple panic catcher functionality in this PR and create another PR for merging the more complex/long term changes into `protocol-v4`.

Reply:
I have updated this PR accordingly and opened an another PR with the long term solution to be merged into newly created `protocol-v4` branch. https://github.com/1Password/shell-plugins/pull/149